pkg/node/stage_unstage.go

```diff
@@ -154,7 +154,6 @@ func (n *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstage
 		}
 	}
 
-	vol.Status.HostPath = ""
```

Comment:
As we were unset-ing the HostPath here, `os.RemoveAll(hostPath)` on deletion becomes in-effective. 

This fix would clean up the host path on volume deletion.

Reply:
Nice catch.